types/stripe/index.d.ts

```diff
@@ -4785,8 +4795,7 @@ declare namespace Stripe {
              * you created a charge with the capture option set to false. Uncaptured payments expire exactly seven days after they are
              * created. If they are not captured by that point in time, they will be marked as refunded and will no longer be capturable.
              */
-            capture(id: string, options: HeaderOptions, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
-            capture(id: string, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
+            capture(id: string, data?: charges.IChargeCaptureOptions, options?: HeaderOptions, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
```

Comment:
Hmm, I think that this needs to be specified something like this:

```typescript
capture(id: string, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
capture(id: string, data: charges.IChargeCaptureOptions, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
capture(id: string, options: HeaderOptions, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
capture(id: string, data: charges.IChargeCaptureOptions, options: HeaderOptions, response?: IResponseFn<charges.ICharge>): Promise<charges.ICharge>;
```

since all of this calls are okay:

```js
capture(id)
capture(id, (err) => {})
capture(id, data)
capture(id, data, (err) => {})
capture(id, options)
capture(id, options, (err) => {})
capture(id, data, options)
capture(id, data, options, (err) => {})
```

note that the Stripe client differs `data` from `options` by looking at the keys in the object

any thoughts @lauvigne, @DanielRosenwasser?

Reply:
Yes you are right, it's better!